http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -0,0 +1,170 @@
+package com.snowplowanalytics.snowplow.collectors.scalastream
+
+import java.util.UUID
+
+import scala.collection.JavaConverters._
+
+import cats.effect.Sync
+import cats.implicits._
+
+import org.http4s.{Request, RequestCookie, Response}
+import org.http4s.Status._
+
+import org.typelevel.ci._
+
+import com.snowplowanalytics.snowplow.CollectorPayload.thrift.model1.CollectorPayload
+
+import com.snowplowanalytics.snowplow.collectors.scalastream.model._
+
+trait Service[F[_]] {
+  def cookie(
+    queryString: Option[String],
+    body: F[Option[String]],
+    path: String,
+    cookie: Option[RequestCookie],
+    userAgent: Option[String],
+    refererUri: Option[String],
+    hostname: F[Option[String]],
+    ip: Option[String],
+    request: Request[F],
+    pixelExpected: Boolean,
+    doNotTrack: Boolean,
+    contentType: Option[String] = None,
+    spAnonymous: Option[String] = None
+  ): F[Response[F]]
+  def determinePath(vendor: String, version: String): String
```

Comment:
It seems to me there is an alternative implementation:

- Remove `determinePath` from this trait.
- `cookieRoutes` invokes `collectorService.cookie(...)` using the original path.
- The collector service determines the fixed path from the original path.

As for which one is better.....  we should consider the tests.  Why is `Service` defined as a trait in the fist place?  I believe the answer is so that we can mock it during tests.  A unit test of `CollectorRoutes` using a mocked `Service` is simpler if the `Service` trait definition is minimized to the essentials.

This is a matter of taste though, so feel free to disagree.

Reply:
`cookie` function is used with [pixel endpoints](https://github.com/snowplow/stream-collector/blob/master/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorRoute.scala#L107) as well in the akka collector. And pixel tracking doesn't use `determinePath` function. Therefore, it isn't included on `cookie` function in there. I tried to follow same structure. If you think it is okay to use `determinePath` with pixel endpoints as well, I guess we can remove it from `Service` trait.

Reply:
OK let's leave it like it is.  Part of me still thinks we haven't picked the right abstractions for the collector.  But on the other hand we know this way will work, and I shouldn't make this task more complicated than it already is!!!